File: 4-7-2024/reduce_string.py
input_string = input("Enter the string: ")

res= ""
i= 0
length= len(input_string)

while i < length:
    count = 1
    while i + 1 < length and input_string[i] == input_string[i + 1]:
        i += 1
        count += 1
    res+= input_string[i]
    if count > 1:
        res+= str(count)
    i += 1

print(res)

File: 1-7-2024/Reduce_string.cpp
#include <iostream>
#include <string>
using namespace std;

string reduceString(const string& str) {
    string result;
    int n = str.length();

    for (int i=0; i<n; ) {
        char current_char=str[i];
        int count = 0;

        while (i<n && str[i]==current_char) {
            count++;
            i++;
        }

        result+=current_char;
        if (count>1) {
            result += to_string(count);
        }
    }
    return result;
}

int main() {

    string input1= "aabbbbeeeeffggg";
    string output1= reduceString(input1);
    cout << "Input: " << input1 << endl;
    cout << "Output: " << output1 << endl;

    string input2= "abbccccc";
    string output2= reduceString(input2);
    cout << "Input: " << input2 << endl;
    cout << "Output: " << output2 << endl;

    return 0;
}
